examples/ping-pong/server.py

```diff
@@ -0,0 +1,43 @@
+"""
+    This server serves back messages sent to `*.echo.identified`
+    The reply will contain the name of the current user
+
+    $ python scripts/nats-pub.py --request user_ulid.echo.identified
+
+"""
+
+from loguru import logger
+
+from kova.current_user import CurrentUser
+from kova.protocol.pingpong_pb2 import PingRequest
+from kova.server import Server, Router
+from kova.message import Publish
+
+
+router = Router()
+
+
+@router.subscribe("*.ping")
+async def ping_pong(msg: PingRequest, current_user: CurrentUser):
+    if msg.first:
```

Comment:
What is the point of `first`? Does each packet has to carry the information whether they are the first? That does not seem very robust as a client could ignore or misuse that field, and it would derail your server-side checks, no?
Is this a common algorithm to filter broadcast duplicates?

Reply:
Changed it to have the server identifying if it's the original message or the duplicate instead of having a marker directly on the message. I saw that solution in a blog but indeed since the message fields are accessible by the clients and can be easily modified it is not a very good solution.